### ISSUE TITLE
RSAAuthentication deprecated

### ISSUE THREAD

**Issue by loodvn** [pexpect/pexpect#717]
Hi there,

I'm getting the following warnings when using pxssh.login:

`command-line line 0: Unsupported option "rsaauthentication"`

My understanding is openSSH (https://www.openssh.com/releasenotes.html) and ssh_config (https://man.openbsd.org/ssh_config) dropped support for RSA authentication for quite some time. 

It seems to come from whenever SSH_OPTS is used (https://github.com/pexpect/pexpect/blob/acb017a97332c19a9295660fe87316926a8adc55/pexpect/pxssh.py#L146), e.g. when self.force_password is True:
https://github.com/pexpect/pexpect/blob/2be6c4d1aa2b9b522636342c2fd54b73c058060d/pexpect/pxssh.py#L321 

Is that option still necessary or is PubkeyAuthentication=no sufficient?

Explanation of the deprecation that provided context for me: https://levelup.gitconnected.com/demystifying-ssh-rsa-in-openssh-deprecation-notice-22feb1b52acd (from https://askubuntu.com/a/1407251)

**Comment by Red-M** [comment]
Thanks but its recommended to move away from pxssh as per the notice in the documentation [here](https://pexpect.readthedocs.io/en/stable/api/pxssh.html#pxssh-control-an-ssh-session).

I can't find the reference in the openssh's release notes that say they dropped `RSAAuthentication` support. I assume this might be a per distro option?

I'm happy to review a PR to remove the option but you are able to subclass `pxssh.pxssh()` to change it as a "monkey patch"

**Comment by loodvn** [comment]
Yeah I've struggled to get the precise note of the dropped support... Also I know veeeery little about OpenSSH/SSH haha, just trying to help out if there's a way to reduce users' confusion from this warning.

It seems from https://unix.stackexchange.com/questions/337774/deprecated-options-when-restarting-openssh-in-stretch and https://community.centminmod.com/threads/centos-7-4-sshd-deprecated-rsaauthentication.13137/ that after OpenSSH version 7.3+ they dropped support for SSH v1, which included the `RSAAuthentication` option.

And the last time I saw it in the ssh_config docs was here with OpenBSD-6.0:
[RSAAuthentication](https://man.openbsd.org/OpenBSD-6.0/sshd_config#RSAAuthentication)
Specifies whether pure RSA authentication is allowed. The default is “yes”. This option applies to protocol version 1 only. 

So I guess since SSH v1 is so old there's no harm in removing the option right? If so I'll put in a PR 👍
